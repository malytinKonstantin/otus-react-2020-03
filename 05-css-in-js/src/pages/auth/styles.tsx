import styled from '@emotion/styled'
import { css } from '@emotion/core'

export const Row = styled.div`
  margin-bottom: 15px;
`

export const Wrapper = styled.div`
  margin: 0 auto;
  padding-top: 100px;
  width: 200px;
`

export const Label = styled.label`
  display: block;
`

export const Input = styled.input`
  width: 100%;
`
