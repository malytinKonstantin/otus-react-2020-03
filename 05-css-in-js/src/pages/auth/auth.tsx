import React, { FC, useEffect } from 'react'
import { Formik, Form, Field } from 'formik'
import { Row, Wrapper, Label, Input } from './styles'

interface AuthPageProps {
  onLogin: Function
}

export const AuthPage: FC<AuthPageProps> = ({ onLogin }) => {
  const initialValues = {
    login: '',
    password: '',
  }

  const handleSubmit = (values: typeof initialValues, form: any) => {
    onLogin(values)
    form.resetForm()
  }

  return (
    <Formik onSubmit={handleSubmit} initialValues={initialValues}>
      {(formProps) => (
        <Form>
          <Wrapper>
            <Row>
              <Label htmlFor="login">name</Label>
              <Field
                type="text"
                name="login"
                id="login"
                component={Input}
                value={formProps.values.login}
                onChange={formProps.handleChange}
              />
            </Row>
            <Row>
              <Label htmlFor="password">password</Label>
              <Field
                type="password"
                name="password"
                id="password"
                component={Input}
                value={formProps.values.password}
                onChange={formProps.handleChange}
              />
            </Row>
            <button type="submit">login</button>
          </Wrapper>
        </Form>
      )}
    </Formik>
  )
}
