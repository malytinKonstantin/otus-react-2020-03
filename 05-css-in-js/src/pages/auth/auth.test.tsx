import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { AuthPage } from './auth'

describe('<AuthPage />', () => {
  it('renders snapshot', () => {
    expect(
      renderer.create(<AuthPage onLogin={jest.fn} />).toJSON(),
    ).toMatchSnapshot()
  })
})
