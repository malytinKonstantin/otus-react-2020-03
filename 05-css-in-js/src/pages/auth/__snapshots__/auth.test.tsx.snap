// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthPage /> renders snapshot 1`] = `
.emotion-12 {
  margin: 0 auto;
  padding-top: 100px;
  width: 200px;
}

.emotion-4 {
  margin-bottom: 15px;
}

.emotion-0 {
  display: block;
}

.emotion-2 {
  width: 100%;
}

<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="emotion-12 emotion-13"
  >
    <div
      className="emotion-4 emotion-5"
    >
      <label
        className="emotion-0 emotion-1"
        htmlFor="login"
      >
        name
      </label>
      <input
        className="emotion-2 emotion-3"
        form={
          Object {
            "dirty": false,
            "errors": Object {},
            "getFieldHelpers": [Function],
            "getFieldMeta": [Function],
            "getFieldProps": [Function],
            "handleBlur": [Function],
            "handleChange": [Function],
            "handleReset": [Function],
            "handleSubmit": [Function],
            "initialErrors": Object {},
            "initialStatus": undefined,
            "initialTouched": Object {},
            "initialValues": Object {
              "login": "",
              "password": "",
            },
            "isSubmitting": false,
            "isValid": true,
            "isValidating": false,
            "registerField": [Function],
            "resetForm": [Function],
            "setErrors": [Function],
            "setFieldError": [Function],
            "setFieldTouched": [Function],
            "setFieldValue": [Function],
            "setFormikState": [Function],
            "setStatus": [Function],
            "setSubmitting": [Function],
            "setTouched": [Function],
            "setValues": [Function],
            "status": undefined,
            "submitCount": 0,
            "submitForm": [Function],
            "touched": Object {},
            "unregisterField": [Function],
            "validateField": [Function],
            "validateForm": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validateOnMount": false,
            "values": Object {
              "login": "",
              "password": "",
            },
          }
        }
        id="login"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="emotion-4 emotion-5"
    >
      <label
        className="emotion-0 emotion-1"
        htmlFor="password"
      >
        password
      </label>
      <input
        className="emotion-2 emotion-3"
        form={
          Object {
            "dirty": false,
            "errors": Object {},
            "getFieldHelpers": [Function],
            "getFieldMeta": [Function],
            "getFieldProps": [Function],
            "handleBlur": [Function],
            "handleChange": [Function],
            "handleReset": [Function],
            "handleSubmit": [Function],
            "initialErrors": Object {},
            "initialStatus": undefined,
            "initialTouched": Object {},
            "initialValues": Object {
              "login": "",
              "password": "",
            },
            "isSubmitting": false,
            "isValid": true,
            "isValidating": false,
            "registerField": [Function],
            "resetForm": [Function],
            "setErrors": [Function],
            "setFieldError": [Function],
            "setFieldTouched": [Function],
            "setFieldValue": [Function],
            "setFormikState": [Function],
            "setStatus": [Function],
            "setSubmitting": [Function],
            "setTouched": [Function],
            "setValues": [Function],
            "status": undefined,
            "submitCount": 0,
            "submitForm": [Function],
            "touched": Object {},
            "unregisterField": [Function],
            "validateField": [Function],
            "validateForm": [Function],
            "validateOnBlur": true,
            "validateOnChange": true,
            "validateOnMount": false,
            "values": Object {
              "login": "",
              "password": "",
            },
          }
        }
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      login
    </button>
  </div>
</form>
`;
